source/adios2/toolkit/query/Query.h

```diff
@@ -93,16 +93,38 @@ class RangeTree
     std::vector<RangeTree> m_SubNodes;
 }; // class RangeTree
 
+struct BlockHit
+{
+    BlockHit(size_t id);
+    BlockHit(size_t id, Box<Dims> &box);
+    BlockHit(const BlockHit &cpy);
+
+    size_t m_ID;
+
+    // if no sublocks, m_Regions is start/count of block (if global array). size=1
+    // with subblocks,
+    //    if global array, m_Regions has all the touched sub blocks with (abs) start count
+    //    if local array, because client needs to read whole block, subblocks is ignored
+    //       size=0
+    // items in this vector are assumed to have no intersection.
+    std::vector<Box<Dims>> m_Regions;
+
+    bool isLocalArrayBlock() const { return (0 == m_Regions.size()); }
+    bool applyIntersection(const BlockHit &tmp);
+    bool applyExtension(const BlockHit &tmp);
+};
+
 class QueryBase
 {
 public:
     virtual ~QueryBase(){};
     virtual bool IsCompatible(const adios2::Box<adios2::Dims> &box) = 0;
     virtual void Print() = 0;
     virtual void BlockIndexEvaluate(adios2::core::IO &, adios2::core::Engine &,
-                                    std::vector<Box<Dims>> &touchedBlocks) = 0;
+                                    // std::vector<Box<Dims>> &touchedBlocks) = 0;
```

Comment:
One extra commented out line here that could be deleted...

Reply:
addressed